result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -0,0 +1,118 @@
+package com.github.kittinunf.result
+
+inline fun <reified X> Result<*, *>.getAs() = when (this) {
+    is Result.Success -> value as? X
+    is Result.Failure -> error as? X
+}
+
+fun <V : Any> Result<V, *>.success(f: (V) -> Unit) = fold(f, {})
+
+fun <E : Exception> Result<*, E>.failure(f: (E) -> Unit) = fold({}, f)
+
+infix fun <V : Any, E : Exception> Result<V, E>.or(fallback: V) = when (this) {
+    is Result.Success -> this
+    else -> Result.Success(fallback)
+}
+
+infix fun <V : Any, E : Exception> Result<V, E>.getOrElse(fallback: V) = when (this) {
+    is Result.Success -> value
+    else -> fallback
+}
+
+inline fun <V : Any, U : Any, E : Exception> Result<V, E>.map(transform: (V) -> U): Result<U, E> =
+    when (this) {
+        is Result.Success -> Result.Success(transform(value))
+        is Result.Failure -> Result.Failure(error)
+    }
+
+inline fun <V : Any, U : Any, E : Exception> Result<V, E>.flatMap(transform: (V) -> Result<U, E>): Result<U, E> =
+    when (this) {
+        is Result.Success -> transform(value)
+        is Result.Failure -> Result.Failure(error)
+    }
+
+fun <V : Any, E : Exception, E2 : Exception> Result<V, E>.mapError(transform: (E) -> E2) = when (this) {
+    is Result.Success -> Result.Success(value)
+    is Result.Failure -> Result.Failure(transform(error))
+}
+
+fun <V : Any, E : Exception, E2 : Exception> Result<V, E>.flatMapError(transform: (E) -> Result<V, E2>) = when (this) {
+    is Result.Success -> Result.Success(value)
+    is Result.Failure -> transform(error)
+}
+
+fun <V : Any, E : Exception> Result<V, E>.any(predicate: (V) -> Boolean): Boolean = try {
+    when (this) {
+        is Result.Success -> predicate(value)
+        is Result.Failure -> false
+    }
+} catch (ex: Exception) {
+    false
+}
+
+fun <V : Any, U : Any> Result<V, *>.fanout(other: () -> Result<U, *>): Result<Pair<V, U>, *> =
+        flatMap { outer -> other().map { outer to it } }
+
+fun <V : Any, E : Exception> List<Result<V, E>>.lift(): Result<List<V>, E> = fold(Result.success(mutableListOf<V>()) as Result<MutableList<V>, E>) { acc, result ->
```

Comment:
imo it should be named `sequence` as in other FP libraries, I'd expect `lift` to only wrap some value in container